sdk/identity/azure-identity/Troubleshoot.md

```diff
@@ -0,0 +1,223 @@
+| Azure CLI Credential | [Azure CLI Troubleshooting Guide](#troubleshooting-azure-cli-authentication-issues) |
+| Azure Powershell Credential | [Azure Powershell Troubleshooting Guide](#troubleshooting-azure-powershell-authentication-issues) |
+
+## Troubleshooting Environment Credential Authentication Issues
+
+### Credential Unavailable Error
+
+#### Environment variables not configured
+
+The `EnvironmentCredential` supports Service Principal authentication and Username + Password Authentication. To utilize the desired way of authentication via `EnvironmentCredential`, you need to ensure the environment variables below are configured properly and the application is able to read them.
+
+##### Service principal with secret
+
+| Variable Name | Value |
+| --- | --- |
+AZURE_CLIENT_ID | ID of an Azure Active Directory application. |
+AZURE_TENANT_ID |ID of the application's Azure Active Directory tenant. |
+AZURE_CLIENT_SECRET | One of the application's client secrets. |
+
+##### Service principal with certificate
+
+| Variable name | Value |
+| --- | --- |
+AZURE_CLIENT_ID |ID of an Azure Active Directory application. |
+AZURE_TENANT_ID | ID of the application's Azure Active Directory tenant. |
+AZURE_CLIENT_CERTIFICATE_PATH | Path to a PEM-encoded certificate file including private key (without password protection). |
+
+##### Username and password
+
+| Variable name | Value |
+| --- | --- |
+AZURE_CLIENT_ID | ID of an Azure Active Directory application. |
+AZURE_USERNAME | A username (usually an email address). |
+AZURE_PASSWORD | The associated password for the given username. |
+
+### Client Authentication Error
+
+The `EnvironmentCredential` supports Service Principal authentication and Username + Password Authentication.
+Please follow the troubleshooting guidelines below for the respective authentication which you tried and failed.
+
+| Authentication Type | Troubleshoot Guide |
+| --- | --- |
+| Service Principal | [Service Principal Auth Troubleshooting Guide](#troubleshooting-username-password-authentication-issues) |
+| Username Password | [Username Password Auth Troubleshooting Guide](#troubleshooting-username-password-authentication-issues) |
+
+## Troubleshooting Username Password Authentication Issues
+
+### Two Factor Authentication Required Error
+
+The `UsernamePassword` credential works only for users whose two factor authentication has been disabled in Azure Active Directory. You can change the Multi Factor Authentication in Azure Portal by following the steps [here](https://docs.microsoft.com/azure/active-directory/authentication/howto-mfa-userstates#change-the-status-for-a-user).
+
+## Troubleshooting Service Principal Authentication Issues
+
+### Illegal/Invalid Argument Issues
+
+#### Client Id
+
+The Client Id is the application Id of the registered application / service principal in Azure Active Directory.
```

Comment:
nit: I would lower-case "client" to align with "tenant" and "client secret", and would capitalize "ID" here and elsewhere
```suggestion
The client ID is the application ID of the registered application / service principal in Azure Active Directory.
```